arangod/VocBase/Methods/Collections.cpp

```diff
@@ -263,14 +264,36 @@ Result Collections::create(TRI_vocbase_t& vocbase,
                info.collectionType != TRI_col_type_e::TRI_COL_TYPE_EDGE) {
       return TRI_ERROR_ARANGO_COLLECTION_TYPE_INVALID;
     }
+
     TRI_ASSERT(info.properties.isObject());
     helper.clear();
     helper.openObject();
     helper.add(arangodb::StaticStrings::DataSourceType,
                arangodb::velocypack::Value(static_cast<int>(info.collectionType)));
     helper.add(arangodb::StaticStrings::DataSourceName,
                arangodb::velocypack::Value(info.name));
+
+    //auto shardingSlice = info.properties.get(StaticStrings::Sharding);
+    auto replicationFactorSlice = info.properties.get(StaticStrings::ReplicationFactor);
+    if(replicationFactorSlice.isNone()) {
+      auto factor = vocbase.replicationFactor();
+      if(factor > 0 && vocbase.IsSystemName(info.name)) {
+        // system collections should at least be replicated twice
+        factor = std::max(vocbase.replicationFactor(), 2u);
+      }
+      helper.add(StaticStrings::ReplicationFactor, VPackValue(vocbase.replicationFactor()));
+    }
+
+    // system collections will be shareded normally - we avoid a self reference when creating _graphs
+    if(vocbase.sharding() == "single" && !vocbase.IsSystemName(info.name)) {
+      auto distribute = info.properties.get(StaticStrings::DistributeShardsLike);
+      if(distribute.isNone()) {
+        helper.add(StaticStrings::DistributeShardsLike, VPackValue(StaticStrings::GraphCollection));
+      }
+    }
+
     helper.close();
+
     VPackBuilder merged =
         VPackCollection::merge(info.properties, helper.slice(), false, true);
```

Comment:
Won't this order of merging lead to user-defined values in `info` being overwritten by the values in `helper`?
So that it won't be possible to set a user-defined replicationFactor and sharding value?

Reply:
Yes but helper contains values provided by info or defaults provided by the vocbase.